appsec/src/extension/ddappsec.c

```diff
@@ -510,7 +509,6 @@ static PHP_FUNCTION(datadog_appsec_push_addresses)
     }
 
     dd_result res = dd_request_exec(conn, &parameters_zv, rasp);
-    zval_ptr_dtor(&parameters_zv);
```

Comment:
This is now sort of correct (you still have an invalid zval, but it has no ill-effects, provided that `dd_request_exec` is not changed in the future in a breaking way), but you chose the worst option. Instead of asking for a Hashtable* and converting it to a zval, you can ask zend_parse_parameters for a zval directly

Reply:
Changed